File: max_sub_array.py
 """
        :type nums: List[int]
        :rtype: int
        """

class Solution(object):
    def maxSubArray(self, nums):
       mid_point = len(nums)/2
       #base case. array is of size 0
       if(len(nums) == 0 or len(nums) == 1):
       	return nums[0]
       else 

File: maxSubArray.java

public class Solution {
	public int[] maxSubArray(int[] nums){
		int maxSoFar;
		int maxEndingHere= 0;

		for(int i = 0; i < nums.length(); i++){
			maxEndingHere = maxEndingHere + nums[i];
			if(maxEndingHere < 0){
				maxEndingHere = 0;
				if(maxEndingHere > maxSoFar){
					maxSoFar = maxEndingHere;
				}
			}
			return maxSoFar;
		}


		}
}